bindings/java/src/main/native/endpoint.cc

```diff
@@ -170,12 +191,21 @@ Java_org_openucx_jucx_ucp_UcpEndpoint_getNonBlockingNative(JNIEnv *env, jclass c
                                                            jlong size, jobject callback,
                                                            jint memory_type)
 {
-    ucs_status_ptr_t request = ucp_get_nb((ucp_ep_h)ep_ptr, (void *)laddr, size,
-                                          raddr, (ucp_rkey_h)rkey_ptr, jucx_request_callback);
+    ucp_request_param_t param = {0};
+
+    jobject jucx_request = jucx_request_allocate(env, callback, &param);
 
+    param.op_attr_mask |= UCP_OP_ATTR_FIELD_MEMORY_TYPE;
+    param.memory_type   = static_cast<ucs_memory_type_t>(memory_type);
+    param.cb.send       = jucx_request_callback;
+
+    ucs_status_ptr_t status = ucp_get_nbx((ucp_ep_h)ep_ptr, (void *)laddr, size,
+                                          raddr, (ucp_rkey_h)rkey_ptr, &param);
+    process_request(env, jucx_request, status);
     ucs_trace_req("JUCX: get_nb request %p, raddr: %zu, size: %zu, result address: %zu",
-                  request, raddr, size, laddr);
-    return process_request(request, callback);
+                  status, raddr, size, laddr);
```

Comment:
need to print debug message before calling process_request
in other places as well

Reply:
unresolved: trace before process

Reply:
unresolved: move trace before process